src/ja/worker/proxy/message_handler.py

```diff
@@ -13,7 +22,27 @@ class WorkerCommandHandler(CommandHandler):
     It starts listening on it for incoming commands.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, worker_client: JobWorker) -> None:
         """!
         Creates the socket and listens on it.
         """
+        self._worker_Client = worker_client
+
+    _server_commands = {
+        "CancelJobCommand": CancelJobCommand,
+        "GetWorkloadCommand": GetWorkloadCommand,
+        "PauseJobCommand": PauseJobCommand,
+        "ChangeFiltersCommand": ChangeFiltersCommand,
+        "StartJobCommand": StartJobCommand
+    }
+
+    def _execute_command(self, command: WorkerCommand) -> Dict[str, object]:
+        return command.execute(self._worker_Client).to_dict()
+
+    def _process_message(self, command_dict: Dict[str, object], type_name: str, user: str) -> Dict[str, object]:
```

Comment:
_process_message (or main_loop as it was renamed by Ilia) needs to verify user permissions.
For the worker client this means checking if the user is part of the admin group (as was done in the server main class).

Reply:
I m not sure why the checking of the user is happening on the worker when it can be detected while handling his command on the server.

Reply:
You have to check permissions on the worker because otherwise any user logged into the worker could directly call the remote and send any sort of command to the worker even if such a user is not an administrator.
Checking permissions on the worker has nothing to do with the human user specified in the job.
You only need to check whether the user sending a command to the worker is part of the admin group (the server will use an admin account to log in to the worker and send the command).

Reply:
You can see the `user_is_admin` function from the server. I think you can move it to the base CommandHandler so that we don't duplicate it.